unittests/unit/test_MarkerFitter.cpp

```diff
@@ -7423,6 +7675,193 @@ TEST(MarkerFitter, FULL_KINEMATIC_RAJAGOPAL)
+  std::vector<std::map<std::string, Eigen::Vector3s>> markerTimesteps;
+  std::vector<std::map<std::string, Eigen::Vector3s>> accTimesteps;
+  std::vector<std::map<std::string, Eigen::Vector3s>> gyroTimesteps;
+  std::vector<bool> newClip;
+  for (int t = 0; t < limitTimesteps; t++)
+  {
+    markerTimesteps.push_back(trc.markerTimesteps[t]);
+    accTimesteps.push_back(imuData.accReadings[t]);
+    gyroTimesteps.push_back(imuData.gyroReadings[t]);
+    newClip.push_back(t == 0);
+  }
+
+  MarkerInitialization init;
+  AccelerationSmoother smoother(poses.cols(), 1e3, 1e-4, false, false);
+  init.poses = smoother.smooth(poses);
+
+  EXPECT_TRUE(testIMUGradients(
+      fitter,
+      accTimesteps,
+      gyroTimesteps,
+      0.005,
+      init,
+      0,
+      accTimesteps.size()));
+}
+#endif
+
+#ifdef ALL_TESTS
+TEST(MarkerFitter, CARMAGO_FINE_TUNE_ON_IMUS)
+{
+  OpenSimFile carmago = OpenSimParser::parseOsim(
+      "dart://sample/grf/CarmagoTest/Models/final.osim");
+  carmago.skeleton->autogroupSymmetricSuffixes();
+  carmago.skeleton->setGravity(Eigen::Vector3s(0, -9.81, 0));
+
+  // Get the raw marker trajectory data
+  OpenSimIMUData imuData = OpenSimParser::loadIMUFromCSV(
+      "dart://sample/grf/CarmagoTest/IMU/treadmill_01_01.csv", true);
+
+  OpenSimTRC trc = OpenSimParser::loadTRC(
+      "dart://sample/grf/CarmagoTest/MarkerData/treadmill_01_01.trc");
+
+  OpenSimMot mot = OpenSimParser::loadMot(
+      carmago.skeleton,
+      "dart://sample/grf/CarmagoTest/IK/treadmill_01_01_ik.mot");
+
+  MarkerFitter fitter(carmago.skeleton, carmago.markersMap);
+  fitter.setTrackingMarkers(carmago.trackingMarkers);
+
+  SensorMap imus;
+  // foot, shank, thigh, trunk
+  imus["foot"] = std::make_pair(
+      carmago.skeleton->getBodyNode("calcn_r"), Eigen::Isometry3s::Identity());
+  imus["shank"] = std::make_pair(
+      carmago.skeleton->getBodyNode("tibia_r"), Eigen::Isometry3s::Identity());
+  imus["thigh"] = std::make_pair(
+      carmago.skeleton->getBodyNode("femur_r"), Eigen::Isometry3s::Identity());
+  imus["trunk"] = std::make_pair(
+      carmago.skeleton->getBodyNode("torso"), Eigen::Isometry3s::Identity());
+
+  fitter.setImuMap(imus);
+
+  int limitTimesteps = 1000;
+  Eigen::MatrixXs poses
+      = mot.poses.block(0, 0, mot.poses.rows(), limitTimesteps);
+  std::vector<std::map<std::string, Eigen::Vector3s>> markerTimesteps;
+  std::vector<std::map<std::string, Eigen::Vector3s>> accTimesteps;
+  std::vector<std::map<std::string, Eigen::Vector3s>> gyroTimesteps;
+  std::vector<bool> newClip;
+  for (int t = 0; t < limitTimesteps; t++)
+  {
+    markerTimesteps.push_back(trc.markerTimesteps[t]);
+    accTimesteps.push_back(imuData.accReadings[t]);
+    gyroTimesteps.push_back(imuData.gyroReadings[t]);
+    newClip.push_back(t == 0);
+  }
+
+  MarkerInitialization init;
+  AccelerationSmoother smoother(poses.cols(), 1e3, 1e-4, true, true);
+  smoother.setIterations(100000);
```

Comment:
Does this take a long time to run in practice?

Reply:
It seems to get done fairly quickly even on long Carmago trials when running on my laptop, but if it ends up being a problem we should swap it out. The trouble is, because the problem is now so stiff, if we go to smaller numbers of iterations we get bad results :/